base/lttagging.dtx

```diff
@@ -0,0 +1,789 @@
+%
+% \DescribeMacro\UseTaggingSocket
+% \DescribeMacro\tag_socket_use:n
+% \DescribeMacro\tag_socket_use:nn
+%  To support tagging in packages we use sockets with names starting
+%  with \texttt{tagsupport/}. Usually, these sockets have exactly two
+%  plugs defined:
+%  \plug{noop} (when no tagging is requested or tagging is not wanted
+%  for some reason) and a second plug that enables the tagging. There
+%  may be more, e.g., tagging with special debugging, etc., but right
+%  now it is usually just on or off.
+%
+%  Given that we sometimes have to suspend tagging, it would be fairly
+%  inefficient to put different plugs into these sockets whenever that
+%  happens. We therefore offer \cs{UseTaggingSocket} which is like
+%  \cs{UseSocket} except that the socket name is specified without
+%  \texttt{tagsupport/}, i.e.,
+% \begin{quote}
+%   \verb=\UseTaggingSocket{foo}=      $\to$
+%   \verb=\UseSocket{tagsupport/foo}=
+% \end{quote}
+%  Beside being slightly shorter, the big advantage is that this way
+%  we can change \cs{UseTaggingSocket} to do nothing when tagging is
+%  suspended with \cs{SuspendTagging} instead of changing the plugs of
+%  the tagging support sockets back and forth.
+%
+%  It is possible to use the tagging support sockets with
+%  \cs{UseSocket} directly, but in this case the socket remains active
+%  if \cs{SuspendTagging} is in force. There my be reasons for doing
+%  that but in general we expect to always use \cs{UseTaggingSocket}.
+%
+%  The L3 programming layer versions \cs{tag_socket_use:n} and
+%  \cs{tag_socket_use:nn} are slightly more efficient than
+%  \cs{UseTaggingSocket} because they do not have to determine how
+%  many arguments the socket takes when disabling it, so in code that
+%  is using the L3 progamming layer we recomment to use them instead
+%  of the CamelCase command.
+%                
+%
+%
+%  \begin{macro}{\SuspendTagging,\ResumeTagging}
+%    
+%    In the kernel, these two commands get dummy definitons so that
+%    they can be used without harm in packages. The real definition is
+%    used when tagging gets enabled.
+%    \begin{macrocode}
+\cs_new_eq:NN \SuspendTagging \use_none:n  
+\cs_new_eq:NN \ResumeTagging  \use_none:n
+%    \end{macrocode}
+%
+%    The correct definition in \pkg{tagpdf} will look different. This
+%    one here is just a simplification for now.
+%    \begin{macrocode}
+\AddToHook{begindocument/before}{
+  \cs_if_exist:NT \tag_stop:n
+  { 
+    \cs_set:Npn \SuspendTagging #1 {
+      \tag_stop:n {#1} 
+%    \end{macrocode}%
+%    Next line is is incomplete (only disables table sockets) and
+%    should instead be done in the above command instead of disabling them
+%    individually because that will otherwise forever change if new
+%    sockets get used.
+%    \begin{macrocode}
+      \__tag_tbl_disable:
+    }
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+    \cs_set:Npn \ResumeTagging #1 {
+      \tag_start:n {#1}
+      \__tag_tbl_enable:
+    }
```

Comment:
and here similar

Reply:
at this point in time `\tag_start:n` doesn't handle the sockets so I had to do something for them. All of these are only stubs that tagpdf should provide real definitions for. Once they are there non of this other than the dummy kernel definitions would remain.

Reply:
What do you mean? The sockets in the table module are set with `\UseTaggingSockets` so enabling/disabling should simply work with \tagstart/\tagstop without changing the plugs.

Reply:
I mean that my version doesn't do that yet ... maybe yours does already, and once it does generally this code should be removed.

Reply:
hm I'm missing something here. Your code has `\UseTaggingSocket` everywhere and if the definition is correct it should simply work ;-). I will have to test that.

Reply:
simply \tag_start/\tag_stop works fine for me in your branch. And one could also remove the `\tl_put_right:Nn \@kernel@before@head {\__tag_tbl_disable:}` in the table code: as the header and footer contain a tag_stop tabular tagging is disabled automatically too.

Reply:
you are right, I thought that more is necessary, but it isn't